engine/src/main/java/io/camunda/zeebe/engine/state/immutable/ElementInstanceState.java

```diff
@@ -58,4 +58,12 @@ void forEachChild(
    * @return a list of process instance keys
    */
   List<Long> getProcessInstanceKeysByDefinitionKey(final long processDefinitionKey);
+
+  /**
+   * Verifies if there are running instances for a given process definition
+   *
+   * @param processDefinitionKey the key of the process definition
+   * @return a boolean indicating if there are running instances
+   */
+  boolean hasRunningInstances(long processDefinitionKey);
```

Comment:
🔧 I think it would also be better to use `active instances` internally. This is more uniform.